frontend/src/components/RoutingFeeDialog.tsx

```diff
@@ -0,0 +1,118 @@
+import { ExternalLinkIcon } from "lucide-react";
+import React from "react";
+import ExternalLink from "src/components/ExternalLink";
+import { Input } from "src/components/ui/input";
+import { Label } from "src/components/ui/label";
+import { toast } from "src/components/ui/use-toast";
+import { useChannels } from "src/hooks/useChannels";
+import { useCSRF } from "src/hooks/useCSRF";
+import { Channel, UpdateChannelRequest } from "src/types";
+import { request } from "src/utils/request";
+import {
+  AlertDialogAction,
+  AlertDialogCancel,
+  AlertDialogContent,
+  AlertDialogDescription,
+  AlertDialogFooter,
+  AlertDialogHeader,
+  AlertDialogTitle,
+} from "./ui/alert-dialog";
+
+type Props = {
+  channel: Channel;
+};
+
+export function RoutingFeeDialog({ channel }: Props) {
+  const currentFee: number = React.useMemo(() => {
+    return Math.floor(channel.forwardingFeeBaseMsat / 1000);
+  }, [channel.forwardingFeeBaseMsat]);
+  const [forwardingFee, setForwardingFee] = React.useState(
+    currentFee ? currentFee.toString() : ""
+  );
+  const { mutate: reloadChannels } = useChannels();
+  const { data: csrf } = useCSRF();
+
+  async function updateFee() {
+    try {
+      if (!csrf) {
+        throw new Error("csrf not loaded");
+      }
+
+      const forwardingFeeBaseMsat = +forwardingFee * 1000;
+
+      console.info(
+        `🎬 Updating channel ${channel.id} with ${channel.remotePubkey}`
+      );
+
+      await request(
+        `/api/peers/${channel.remotePubkey}/channels/${channel.id}`,
+        {
+          method: "PATCH",
+          headers: {
+            "X-CSRF-Token": csrf,
+            "Content-Type": "application/json",
+          },
+          body: JSON.stringify({
+            forwardingFeeBaseMsat: forwardingFeeBaseMsat,
+          } as UpdateChannelRequest),
+        }
+      );
+
+      await reloadChannels();
+      toast({ title: "Sucessfully updated channel" });
+    } catch (error) {
+      console.error(error);
+      toast({
+        variant: "destructive",
+        description: "Something went wrong: " + error,
+      });
+    }
+  }
+
+  return (
+    <AlertDialogContent>
+      <AlertDialogHeader>
+        <AlertDialogTitle>Update Channel Forwarding Fee</AlertDialogTitle>
+        <AlertDialogDescription>
+          <p className="mb-4">
+            Adjust the fee you charge for each payment passing through your
+            channel.{" "}
```

Comment:
Should we add:

By default we set a high fee to prevent unwanted routing.

Reply:
Not sure if we do this in LND (and I don't think we should override there for other backends), should I add this only for LDK then?

Reply:
Not sure if this looks good, too much of text (maybe we can rearrange, lmk) and it also adds an info call:

<img width="100%" alt="Screenshot 2024-08-09 at 7 04 00 PM" src="https://github.com/user-attachments/assets/98994da3-bbbc-4769-829f-f4581812e5ad">

Reply:
Ah I see.

Maybe we can change the copy to "A high fee (e.g. 100,000 sats) can be set to prevent unwanted routing."

The input should also show the current value, right? in your screenshot it shows an empty value

Reply:
Yes it will, here it's 0 that's why it's empty

Reply:
Looks good, changed (updated screenshot in PR main comment)